packages/compiler/core/checker.ts

```diff
@@ -2136,10 +2136,8 @@ export function createChecker(program: Program): Checker {
       type.namespace?.models.set(type.name, type);
     }
 
-    const inheritedProps = Array.from(walkPropertiesInherited(type));
-
```

Comment:
You had the right idea, not using walkPropertiesInherited for this. It's cleaner, IMHO to search up for the particular overridden property. But walkPropertiesInherited did still need fixing for the other use cases where it's warranted.